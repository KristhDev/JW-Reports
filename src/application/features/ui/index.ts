export { default as uiReducer, UI_INITIAL_STATE, setIsDataExporting, setIsKeyboardVisible, setOldDatetimePicker } from './slice';
export * from './types';