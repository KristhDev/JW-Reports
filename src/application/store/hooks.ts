import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();