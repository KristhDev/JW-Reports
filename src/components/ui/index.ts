export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Title } from './Title';