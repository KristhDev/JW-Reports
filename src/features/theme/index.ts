export { default as themeReducer, setTheme } from './slice';