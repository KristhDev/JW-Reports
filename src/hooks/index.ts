export { default as useAuth } from './useAuth';
export { default as useTheme } from './useTheme';