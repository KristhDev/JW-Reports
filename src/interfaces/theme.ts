export type Theme = 'dark' | 'light' | 'default';

/* Defining the interface for the ThemeState. */
export interface ThemeState {
    colors: Colors;
    deviceTheme: Theme;
    isLoadedTheme: boolean;
    selectedTheme: Theme,
    theme: Theme,
}

/* Defining the interface for the Colors object. */
export interface Colors {
    background: string;
    bottom: string;
    button: string;
    buttonDark: string;
    card: string;
    contentHeader: string;
    focus: string;
    header: string;
    headerText: string;
    icon: string;
    inputText: string;
    linkText: string;
    modal: string;
    modalText: string;
    navbar: string;
    text: string;
    titleSecondary: string;
    titleText: string;
}

/**
 * SetThemePayload is an object with a property called theme that is of type Theme
 * @property {Theme} theme - The theme to set.
 */
export type SetThemePayload = {
    theme: Theme
}