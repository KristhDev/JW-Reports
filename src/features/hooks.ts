import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;