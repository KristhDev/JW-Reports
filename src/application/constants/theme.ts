import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];