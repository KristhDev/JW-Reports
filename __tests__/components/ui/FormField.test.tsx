import React from 'react';
import { act, fireEvent, render, screen } from '@testing-library/react-native';
import { Formik } from 'formik';
import Icon from 'react-native-vector-icons/Ionicons';

/* Components */
import { FormField } from '../../../src/components/ui';

/* Hooks */
import { useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

/* Mock hooks */
jest.mock('../../../src/hooks/useTheme.ts');

const fieldLabel = 'Nombre:';
const fieldPlaceholder = 'Ingrese su nombre';
const textValue = 'Test name';

describe('Test in <FormField /> component', () => {
    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors }
    });

    const renderComponent = () => {
        render(
            <Formik
                initialValues={{ name: '' }}
                onSubmit={ () => {} }
            >
                { () => (
                    <FormField
                        autoCapitalize="none"
                        icon={
                            <Icon
                                color={ darkColors.icon }
                                name="person-outline"
                                size={ 25 }
                                testID="form-field-icon"
                            />
                        }
                        label={ fieldLabel }
                        name="name"
                        placeholder={ fieldPlaceholder }
                    />
                ) }
            </Formik>
        );
    }

    it('should to match snapshot', () => {
        renderComponent();
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render respective props', () => {
        renderComponent();

        /* Get elements with props of component */
        const label = screen.getByTestId('form-field-label');
        const input = screen.getByTestId('form-field-text-input');
        const icon = screen.getByTestId('form-field-icon');

        /* Check if exists this elemets and has props */
        expect(label).toBeTruthy();
        expect(label.props.children).toBe(fieldLabel);
        expect(input).toBeTruthy();
        expect(input.props.autoCapitalize).toBe('none');
        expect(input.props.placeholder).toBe(fieldPlaceholder);
        expect(icon).toBeTruthy();
    });

    it('should change value of text input', async () => {
        renderComponent();

        /* Get text input */
        const input = screen.getByTestId('form-field-text-input');

        await act(() => {

            /* Change value of input */
            fireEvent(input, 'onChangeText', textValue);
        });

        /* Check if change value of input */
        expect(input.props.value).toBe(textValue);
    });
});