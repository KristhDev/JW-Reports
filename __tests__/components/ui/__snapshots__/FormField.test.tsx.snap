// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FormField /> component should to match snapshot 1`] = `
<View
  style={
    {
      "display": "flex",
      "padding": 10,
      "width": 675,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 18,
          "marginBottom": 5,
          "padding": 5,
        },
        undefined,
      ]
    }
    testID="form-field-label"
  >
    Nombre:
  </Text>
  <View
    style={
      {
        "borderColor": "transparent",
        "borderRadius": 7,
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "borderColor": "#FFFFFF",
          "borderRadius": 5,
          "borderWidth": 0.5,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "transparent",
              "borderRadius": 5,
              "borderWidth": 1.5,
              "flexDirection": "row",
              "paddingRight": 10,
            },
            undefined,
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          cursorColor="#C0A7E1"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSelectionChange={[Function]}
          placeholder="Ingrese su nombre"
          placeholderTextColor="#BFBFBF"
          selection={
            {
              "end": 0,
              "start": 0,
            }
          }
          selectionColor="#A0B9E2"
          style={
            [
              {
                "color": "#FFFFFF",
                "flex": 1,
                "fontSize": 17,
                "paddingLeft": 10,
                "paddingRight": 5,
                "textAlignVertical": "center",
              },
              undefined,
            ]
          }
          testID="form-field-text-input"
          value=""
        />
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#BFBFBF",
                "fontSize": 25,
              },
              undefined,
              {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
          testID="form-field-icon"
        >
          ï‘˜
        </Text>
      </View>
    </View>
  </View>
</View>
`;
